api/handler/versioning.go

```diff
@@ -62,7 +65,7 @@ func (h *handler) GetBucketVersioningHandler(w http.ResponseWriter, r *http.Requ
 
 	settings, err := h.obj.GetBucketSettings(r.Context(), bktInfo)
 	if err != nil {
-		if errors.IsS3Error(err, errors.ErrNoSuchBucket) {
+		if errors.IsS3Error(err, errors.ErrNoSuchKey) {
```

Comment:
We still have to return `VersioningConfiguration` even though we haven't settings file. So this shouldn't be changed